From 7fed288641ba8d35b2104daded7f67633a04fdd9 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <Dinh.Nguyen@freescale.com>
Date: Thu, 22 Jan 2009 11:42:47 -0600
Subject: [PATCH] ENGR00106017 Remove USB Device 31-byte data alignment for MX51 TO2

The software workaround to fix a 31-byte data alignment issue is
no longer needed on MX51 TO2.

Signed-off-by: Dinh Nguyen <Dinh.Nguyen@freescale.com>
---
 drivers/usb/gadget/arcotg_udc.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/arcotg_udc.h b/drivers/usb/gadget/arcotg_udc.h
index fc75890..1d130f8 100644
--- a/drivers/usb/gadget/arcotg_udc.h
+++ b/drivers/usb/gadget/arcotg_udc.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -25,7 +25,7 @@
 
 #define MSC_BULK_CB_WRAP_LEN 31
 #define USE_MSC_WR(len) (((cpu_is_mx37_rev(CHIP_REV_1_0) == 1) ||\
-	(cpu_is_mx51() == 1)) && ((len) == MSC_BULK_CB_WRAP_LEN))
+	(cpu_is_mx51_rev(CHIP_REV_2_0) < 0)) && ((len) == MSC_BULK_CB_WRAP_LEN))
 
 /* Iram patch */
 #ifdef CONFIG_USB_STATIC_IRAM_PPH
-- 
1.5.4.4

